ipc/api/src/checkpoint.rs

```diff
@@ -86,3 +86,43 @@ pub struct BottomUpCheckpoint {
     /// The list of messages for execution
     pub msgs: Vec<IpcEnvelope>,
 }
+
+impl Display for BottomUpCheckpoint {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "BottomUpCheckpoint(subnet_id = {}, height = {}, hash = {}, next_config_number = {}, msgs = {})",
+            self.subnet_id,
+            self.block_height,
+            hex::encode(&self.block_hash),
+            self.next_configuration_number,
+            join_to_string(self.msgs.as_slice()),
```

Comment:
Do you really want to log all messages included in a checkpoint? This isn't even the `Debug` format. Wouldn't the number of messages be enough, perhaps?

Reply:
I used them to debug tests, I mean these cli methods should be used for checking errors, they have been handy when system goes wrong. But I think you are right in the sense that JSON might be better format.